pandas/core/dtypes/common.py

```diff
@@ -2016,3 +2016,54 @@ def pandas_dtype(dtype):
         raise TypeError("dtype '{}' not understood".format(dtype))
 
     return npdtype
+
+
+def extract_array(obj, extract_numpy=False):
```

Comment:
move to pandas.core.dtypes.cast (or maybe into internals somewhere)

Reply:
actually i would put it there, all of these routines were recently moved there (like init_dict, prep_ndarray) and so on. they DO need to be moved out to another module (which you can create if you'd like), e.g. pandas.core.internals.arrays would be ok

Reply:
I'll throw it in constructors for now (this is probably we're it'll be used most often).

Reply:
Nevermind. That's causing all sorts of circular import issues.

Reply:
pls move this, internals/blocks.py if needed. it does *not* belong here.

Reply:
Moved.

This required inlining the import in arrays/category and arrays/numpy_, since internals imports from arrays.